tests/models/ernie_m/test_modeling_ernie_m.py

```diff
@@ -0,0 +1,306 @@
+        )
+        if is_torch_available()
+        else ()
+    )
+    all_generative_model_classes = ()
+
+    def setUp(self):
+        self.model_tester = ErnieMModelTester(self)
+        self.config_tester = ConfigTester(self, config_class=ErnieMConfig, hidden_size=37)
+
+    def test_config(self):
+        self.config_tester.run_common_tests()
+
+    def test_model(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_model(*config_and_inputs)
+
+    def test_model_various_embeddings(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        for type in ["absolute", "relative_key", "relative_key_query"]:
+            config_and_inputs[0].position_embedding_type = type
+            self.model_tester.create_and_check_model(*config_and_inputs)
+
+    def test_for_multiple_choice(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_for_multiple_choice(*config_and_inputs)
+
+    def test_for_question_answering(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_for_question_answering(*config_and_inputs)
+
+    def test_for_sequence_classification(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_for_sequence_classification(*config_and_inputs)
+
+    def test_for_uiem(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_for_uiem(*config_and_inputs)
+
+    def test_for_token_classification(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_for_token_classification(*config_and_inputs)
+
+    @slow
+    def test_model_from_pretrained(self):
+        for model_name in ERNIE_M_PRETRAINED_MODEL_ARCHIVE_LIST[:1]:
+            model = ErnieMModel.from_pretrained(model_name)
+            self.assertIsNotNone(model)
+
+
+@require_torch
+class ErnieMModelIntegrationTest(unittest.TestCase):
+    @slow
+    def test_inference_model(self):
+        model = ErnieMModel.from_pretrained("susnato/ernie-m-base_pytorch")
+        input_ids = torch.tensor([[0, 1, 2, 3, 4, 5]])
+        output = model(input_ids)[0]
+
+        # TODO Replace vocab size
+        hidden_size = 784
+
+        expected_shape = torch.Size((1, 6, hidden_size))
+        self.assertEqual(output.shape, expected_shape)
+
+        expected_slice = torch.tensor(
+            [
+                [
+                    [0.09642269, -0.00556359, -0.03608308],
+                    [0.00521614, 0.09374719, 0.03041629],
+                    [-0.08879025, -0.00807481, -0.12337112],
+                ]
+            ]
+        )
+
+        self.assertTrue(torch.allclose(output[:, :3, :3], expected_slice, atol=1e-3))
```

Comment:
nit: if you feel like the different classes have enough tests, it's fine, but would be cool to make sure that the other models's checkpoints match the expected behavior!